test/mockupdb/test_handshake.py

```diff
@@ -217,5 +220,42 @@ def test_handshake_not_either(self):
         with self.assertRaisesRegex(AssertionError, "does not match"):
             test_hello_with_option(self, OpMsg)
 
+    def test_handshake_max_wire(self):
+        server = MockupDB()
+        primary_response = {"hello": 1, "ok": 1,
+                            "minWireVersion": 0, "maxWireVersion": 6}
+        self.found_auth_msg = False
+
+        def responder(request):
+            if request.matches(OpMsg, saslStart=1):
+                self.found_auth_msg = True
+                # Immediately closes the connection with
+                request.reply(OpMsgReply(**primary_response,
+                                         **{'payload':
+                                                b'r=wPleNM8S5p8gMaffMDF7Py4ru9bnmmoqb0'
+                                                b'1WNPsil6o=pAvr6B1garhlwc6MKNQ93ZfFky'
+                                                b'tXdF9r,'
+                                                b's=4dcxugMJq2P4hQaDbGXZR8uR3ei'
+                                                b'PHrSmh4uhkg==,i=15000',
+                                            "saslSupportedMechs": [
+                                                "SCRAM-SHA-1"]}))
+            else:
+                return request.reply(OpMsgReply(**primary_response))
```

Comment:
This looks like it responds to an OP_QUERY ismaster handshake with an OP_MSG response. Let's change it to use OP_REPLY. Eg:  `request.reply(**primary_response)`

Reply:
Done.